modules/navier_stokes/src/auxkernels/NSComputeLiquidFraction.C

```diff
@@ -0,0 +1,50 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NSComputeLiquidFraction.h"
+#include "MooseMesh.h"
+#include "NS.h"
+
+registerMooseObject("NavierStokesApp", NSComputeLiquidFraction);
+
+InputParameters
+NSComputeLiquidFraction::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+  params.addClassDescription("Computes liquid fraction f_L given the temperature.");
+  params.addRequiredParam<MooseFunctorName>(NS::temperature, "The temperature.");
+  params.addRequiredParam<MooseFunctorName>("T_solidus", "The solidus temperature.");
+  params.addRequiredParam<MooseFunctorName>("T_liquidus", "The liquidus temperature.");
+  return params;
+}
+
+NSComputeLiquidFraction::NSComputeLiquidFraction(const InputParameters & parameters)
+  : AuxKernel(parameters),
+    _T(getFunctor<ADReal>(NS::temperature)),
+    _T_solidus(getFunctor<ADReal>("T_solidus")),
+    _T_liquidus(getFunctor<ADReal>("T_liquidus"))
+{
+}
+
+Real
+NSComputeLiquidFraction::computeValue()
+{
+  const auto elem_arg = makeElemArg(_current_elem);
+
+  if (raw_value(_T_liquidus(elem_arg)) < raw_value(_T_solidus(elem_arg)))
+    paramError("The specified liquidus temperature is smaller than the solidus one.");
```

Comment:
```suggestion
    mooseError("The specified liquidus temperature is smaller than the solidus one.");
```
`paramError` is not thread safe